napari_clusters_plotter/_plotter.py

```diff
@@ -362,36 +362,59 @@ def run(self, properties, plot_x_axis_name, plot_y_axis_name, plot_cluster_name=
         if plot_cluster_name is not None and plot_cluster_name != "label" and plot_cluster_name in list(properties.keys()):
             self.cluster_ids = properties[plot_cluster_name]
 
-            color = ['#ff7f0e', '#1f77b4', '#2ca02c', '#d62728', '#9467bd', '#8c564b', '#e377c2', '#7f7f7f', '#bcbd22',
-                     '#17becf']
+            # get long colormap from function
+            colors = get_nice_colormap()
+            
             self.graphics_widget.pts = self.graphics_widget.axes.scatter(
                 self.data_x,
                 self.data_y,
-                c=[color[int(x)] for x in self.cluster_ids],
+                c=[colors[int(x) % len(colors)] for x in self.cluster_ids],
                 cmap='Spectral',
-                s=10
+                # here spot size is set differentially: larger (5) for all clustered datapoints (id >=0)
+                # and smaller (2.5) for the noise points with id = -1
+                s=[5 if id >= 0 else 2.5 for id in self.cluster_ids],
+
+                # here alpha is set differentially: higher (0.7) for all clustered datapoints (id >= 0)
+                # and lower (0.3) for the noise points with id = -1
+                alpha = [0.7 if id >= 0 else 0.3 for id in self.cluster_ids]
             )
             self.graphics_widget.selector.disconnect()
             self.graphics_widget.selector = SelectFromCollection(self.graphics_widget,self.graphics_widget.axes, 
                                                                  self.graphics_widget.pts)
 
-            cluster_ids_in_space = generate_parametric_cluster_image(self.analysed_layer.data, self.cluster_ids)
+            # measuring time to check if speedup is worth it
+            import time
```

Comment:
should we leave in the code measurement of the process time or was it just for testing?

Reply:
This was just for testing purposes, I will remove it with the next commit.

Reply:
I do not know if you know about profiler and how to use it but if not then you may read a little about it here: https://napari.org/developers/profiling.html and here https://gitlab.mimuw.edu.pl/python-tools/debugger